File: app/src/main/java/com/example/hamzewi/looping/DemoInfiniteAdapter.java
package com.example.hamzewi.looping;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;


import com.asksira.loopingviewpager.LoopingPagerAdapter;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class DemoInfiniteAdapter extends LoopingPagerAdapter<String> {

    private static final int VIEW_TYPE_NORMAL = 100;
    private static final int VIEW_TYPE_SPECIAL = 101;

    public DemoInfiniteAdapter(Context context, ArrayList<String> itemList, boolean isInfinite) {
        super(context, itemList, isInfinite);
    }

    @Override
    protected int getItemViewType(int listPosition) {
        if (itemList.get(listPosition) == "0") return VIEW_TYPE_SPECIAL;
        return VIEW_TYPE_NORMAL;
    }

    @Override
    protected View inflateView(int viewType, ViewGroup container, int listPosition) {
        if (viewType == VIEW_TYPE_SPECIAL) return LayoutInflater.from(context).inflate(R.layout.item_special, container, false);
        return LayoutInflater.from(context).inflate(R.layout.item_pager, container, false);
    }

    @Override
    protected void bindView(View convertView, int listPosition, int viewType) {
        if (viewType == VIEW_TYPE_SPECIAL) return;
        ImageView imageView = (ImageView) convertView.findViewById(R.id.image);
        Glide.with(context).load(itemList.get(listPosition)).into(imageView);
    }


}

File: app/src/main/java/com/example/hamzewi/looping/DemoInfiniteAdapter.kt
package com.example.hamzewi.looping


import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView


import com.asksira.loopingviewpager.LoopingPagerAdapter
import com.bumptech.glide.Glide

import java.util.ArrayList

class DemoInfiniteAdapter(context: Context, itemList: ArrayList<String>, isInfinite: Boolean) : LoopingPagerAdapter<String>(context, itemList, isInfinite) {

    override fun getItemViewType(listPosition: Int): Int {
        return if (itemList[listPosition] === "0") VIEW_TYPE_SPECIAL else VIEW_TYPE_NORMAL
    }

    override fun inflateView(viewType: Int, container: ViewGroup, listPosition: Int): View {
        return if (viewType == VIEW_TYPE_SPECIAL) LayoutInflater.from(context).inflate(R.layout.item_special, container, false) else LayoutInflater.from(context).inflate(R.layout.item_pager, container, false)
    }

    override fun bindView(convertView: View, listPosition: Int, viewType: Int) {
        if (viewType == VIEW_TYPE_SPECIAL) return
        val imageView = convertView.findViewById<View>(R.id.image) as ImageView
        Glide.with(context).load(itemList[listPosition]).into(imageView)
    }

    companion object {

        private val VIEW_TYPE_NORMAL = 100
        private val VIEW_TYPE_SPECIAL = 101
    }


}
